run_test.sh

```diff
@@ -12,14 +12,24 @@
 # Use `export WARPX_TEST_ARCH=CPU` or `export WARPX_TEST_ARCH=GPU` in order
 # to run the tests on CPU or GPU respectively.
 
-# Create test directory
-rm -rf test_dir
-mkdir test_dir
-cd test_dir
+# Remove contents and link to a previous test directory (intentionally two arguments)
+rm -rf test_dir/* test_dir
+# Create a temporary test directory
+tmp_dir=$(mktemp -d -t ci-XXXXXXXXXX)
```

Comment:
`-t ci-XXXXXXXXXX` does not seam to be supported as one would expect on OSX, so we can remove it.